pipelined/bevy_sprite2/src/render/mod.rs

```diff
@@ -146,75 +148,72 @@ impl FromWorld for SpriteShaders {
     }
 }
 
-struct ExtractedSprite {
+pub struct ExtractedSprite {
     transform: Mat4,
     rect: Rect,
     handle: Handle<Image>,
     atlas_size: Option<Vec2>,
-}
-
-#[derive(Default)]
-pub struct ExtractedSprites {
-    sprites: Vec<ExtractedSprite>,
+    vertex_index: usize,
 }
 
 pub fn extract_atlases(
+    mut commands: Commands,
     texture_atlases: Res<Assets<TextureAtlas>>,
-    atlas_query: Query<(&TextureAtlasSprite, &GlobalTransform, &Handle<TextureAtlas>)>,
-    mut render_world: ResMut<RenderWorld>,
+    atlas_query: Query<(
+        Entity,
+        &TextureAtlasSprite,
+        &GlobalTransform,
+        &Handle<TextureAtlas>,
+    )>,
 ) {
-    let mut extracted_sprites = Vec::new();
-    for (atlas_sprite, transform, texture_atlas_handle) in atlas_query.iter() {
+    for (entity, atlas_sprite, transform, texture_atlas_handle) in atlas_query.iter() {
         if !texture_atlases.contains(texture_atlas_handle) {
```

Comment:
I know you didn't touch this code, but this can be shortened to:
```
let texture_atlas = match texture_atlases.get(texture_atlas_handle) {
    None => continue,
    Some(texture_atlas) => texture_atlas,
};

let rect = texture_atlas.textures[atlas_sprite.index as usize];
commands
    .get_or_spawn(entity)
    .insert_bundle((ExtractedSprite {
        atlas_size: Some(texture_atlas.size),
        transform: transform.compute_matrix(),
        rect,
        handle: texture_atlas.texture.clone_weak(),
        vertex_index: 0,
    },));
```
This is also clearer and has less nesting.

Reply:
The first check is totally redundant with the second if statement. I generally favor single nested `if let Some()` over extracting out variables because i think it "flows" better. The texture atlas implementation now matches the normal sprite implementation.